crates/core/tedge/src/cli/mqtt/subscribe.rs

```diff
@@ -82,6 +83,7 @@ fn subscribe(cmd: &MqttSubscribeCommand) -> Result<(), MqttError> {
     }
 
     let (mut client, mut connection) = Client::new(options, DEFAULT_QUEUE_CAPACITY);
+    disconnect_if_interrupted(client.clone());
```

Comment:
Why not do this for the publisher as well, to handle cases like interrupting the publisher that's stuck in the connect phase itself?

Reply:
Why not. A publisher will unlikely be stuck (`tedge mqtt pub` disconnects on error and awaits nothing more than an acknowledgement) but the connection could be not responsive.

=> I will add it for consistency

Reply:
Added by https://github.com/thin-edge/thin-edge.io/pull/2936/commits/531411e2ba94302723dc084b31fbc9b3cfbac523